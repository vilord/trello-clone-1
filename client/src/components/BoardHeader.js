import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';

import './BoardHeader.css';

const getStatusIcon = boardStatus => {
  if (boardStatus === 'Public') {
    return 'globe';
  } else if (boardStatus === 'Team') {
    return 'users';
  }
  return 'lock';
};

const BoardHeader = ({
  title,
  team,
  favorite,
  boardStatus,
  showRenameBoard,
  showAddToTeam,
  showVisibilityMenu,
  showBoardMenu,
}) => {
  const statusIcon = getStatusIcon(boardStatus);
  const favStyle = {
    color: '#f2d600',
    backgroundColor: '#0169a8',
  };
  return (
    <div className="BoardHeader">
      <div>
        <h3 onClick={showRenameBoard}>{title}</h3>
        <Icon name="empty star" style={favorite ? favStyle : null} />
        <p onClick={showAddToTeam}>{team}</p>
        <p onClick={showVisibilityMenu}>
          <Icon name={statusIcon} />
          {boardStatus}
        </p>
      </div>
      <p onClick={showBoardMenu}>
        <Icon name="ellipsis horizontal" />
        Show Menu
      </p>
    </div>
  );
};

BoardHeader.defaultProps = {
  team: 'Personal',
};

BoardHeader.propTypes = {
  title: PropTypes.string.isRequired,
  team: PropTypes.string,
  favorite: PropTypes.bool,
  boardStatus: PropTypes.oneOf(['Public', 'Team', 'Private']).isRequired,
  showRenameBoard: PropTypes.func.isRequired,
  showAddToTeam: PropTypes.func.isRequired,
  showVisibilityMenu: PropTypes.func.isRequired,
  showBoardMenu: PropTypes.func.isRequired,
};

export default BoardHeader;
