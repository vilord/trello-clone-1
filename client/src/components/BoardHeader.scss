.BoardHeader {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 40px;

  > p {
    min-width: 110px;
    margin: 0;
    margin-right: 10px;
    padding: 7px;
    cursor: pointer;
    text-decoration: underline;
    border-radius: 3px;

    i {
      text-decoration: none;
    }

    &:hover {
      background-color: #0169a8;
    }
  }
}

.BoardHeader > div {
  display: flex;
  align-items: center;
  margin-left: 10px;

  h3,
  p {
    margin: 3px;
    padding: 0;
    padding: 5px 10px;
    cursor: pointer;
    border-radius: 3px;

    &:hover {
      background-color: #0169a8;
    }
  }

  > i {
    font-weight: 900;
    width: 30px;
    height: 30px;
    margin: 3px;
    padding-top: 6px;
    cursor: pointer;
    border-radius: 3px;

    &:hover {
      background-color: #0169a8;
    }
  }
}
