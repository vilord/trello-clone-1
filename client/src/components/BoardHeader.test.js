import React from 'react';
import { shallow } from 'enzyme';

import BoardHeader from './BoardHeader';

describe('BoardHeader Component', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  let component;
  let minProps;
  beforeEach(() => {
    minProps = {
      title: 'Trello Clone',
      boardStatus: 'Private',
      showRenameBoard: jest.fn(),
      showAddToTeam: jest.fn(),
      showVisibilityMenu: jest.fn(),
      showBoardMenu: jest.fn(),
    };
    component = shallow(<BoardHeader {...minProps} />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  it('sets the Board title using props.title', () => {
    expect(component.find('h3').text()).toBe(minProps.title);
  });

  it('sets the Board team using props.team', () => {
    // Default value
    expect(
      component
        .find('p')
        .first()
        .text(),
    ).toBe('Personal');
    component.setProps({ team: 'Test Team' });
    expect(
      component
        .find('p')
        .first()
        .text(),
    ).toBe('Test Team');
  });

  it('sets the favorite style when props.favorite = true', () => {
    expect(
      component
        .find('Icon')
        .first()
        .prop('style'),
    ).toBeNull();

    component.setProps({ favorite: true });

    expect(
      component
        .find('Icon')
        .first()
        .prop('style'),
    ).not.toBeNull();
  });

  it('sets the text of the 2nd p element with props.boardStatus', () => {
    expect(
      component
        .find('p')
        .at(1)
        .text(),
    ).toMatch(minProps.boardStatus);
  });

  it('sets lock icon when props.boardStatus = Private', () => {
    expect(
      component
        .find('Icon')
        .at(1)
        .prop('name'),
    ).toBe('lock');
  });

  it('sets globe icon when props.boardStatus = Public', () => {
    component.setProps({ boardStatus: 'Public' });
    expect(
      component
        .find('Icon')
        .at(1)
        .prop('name'),
    ).toBe('globe');
  });

  it('sets users icon when props.boardStatus = Team', () => {
    component.setProps({ boardStatus: 'Team' });
    expect(
      component
        .find('Icon')
        .at(1)
        .prop('name'),
    ).toBe('users');
  });

  it('calls props.showRenameBoard when Board Title is clicked', () => {
    expect(minProps.showRenameBoard).not.toHaveBeenCalled();
    component.find('h3').simulate('click');
    expect(minProps.showRenameBoard).toHaveBeenCalled();
  });

  it('calls props.showAddToTeam when Team is clicked', () => {
    expect(minProps.showAddToTeam).not.toHaveBeenCalled();
    component
      .find('p')
      .first()
      .simulate('click');
    expect(minProps.showAddToTeam).toHaveBeenCalled();
  });

  it('calls props.showVisibilityMenu when Visibility is clicked', () => {
    expect(minProps.showVisibilityMenu).not.toHaveBeenCalled();
    component
      .find('p')
      .at(1)
      .simulate('click');
    expect(minProps.showVisibilityMenu).toHaveBeenCalled();
  });

  it('calls props.showVisibilityMenu when Visibility is clicked', () => {
    expect(minProps.showBoardMenu).not.toHaveBeenCalled();
    component
      .find('p')
      .last()
      .simulate('click');
    expect(minProps.showBoardMenu).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
