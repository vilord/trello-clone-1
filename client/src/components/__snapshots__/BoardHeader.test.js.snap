// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardHeader Component matches snapshot 1`] = `
<div
  className="BoardHeader"
>
  <div>
    <h3
      onClick={[Function]}
    >
      Trello Clone
    </h3>
    <Icon
      as="i"
      name="empty star"
      style={null}
    />
    <p
      onClick={[Function]}
    >
      Personal
    </p>
    <p
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="lock"
      />
      Private
    </p>
  </div>
  <p
    onClick={[Function]}
  >
    <Icon
      as="i"
      name="ellipsis horizontal"
    />
    Show Menu
  </p>
</div>
`;
